easybuild/easyblocks/p/python.py

```diff
@@ -128,6 +132,39 @@ def __init__(self, *args, **kwargs):
             easybuild_subdir = log_path()
             self.pythonpath = os.path.join(easybuild_subdir, 'python')
 
+        ext_defaults = {
+            # The only installed packages at this point are default installed ones, e.g. pip&setuptools
+            # And we want to upgrade them cleanly, i.e. uninstall them
+            'pip_ignore_installed': False,
+            # Python installations must be clean. Requires pip >= 9
+            'sanity_pip_check': LooseVersion(self._get_pip_ext_version() or '0') >= LooseVersion('9'),
+        }
+        # Some older ECs break if we enable download_dep_fail due to missing packages
+        # Since Python 2.7.14 (and all Python 3 versions) we usually enable download_dep_fail in the ECs
+        # already so this should not lead to failures
+        if LooseVersion(self.version) >= LooseVersion('2.7.14'):
+            ext_defaults['download_dep_fail'] = True
```

Comment:
@Flamefire I don't think we should automatically enable this at this point, that's a rather intrusive change in the default behavior, which we should only make in EasyBuild v5.0 (cfr. https://github.com/easybuilders/easybuild-easyblocks/issues/2127).

I fully agree with you that we should do this, which is why the easyconfig test suite has been requiring this for a while now, and I'm looking forward to not having to worry about this anymore and enabling it by default, but we shouldn't for the time being.

We're not enabling `download_dep_fail` in the oldest Python 3 easyconfigs (and it's not worth spending time in trying to enable that there). Those easyconfigs are old enough that they'll be archived in EasyBuild v5.0, so they won't block enabling `download_dep_fail` by default.

If I understand correctly, just removing this part won't affect the rest of this PR, correct me if I'm wrong...

Reply:
Postponed to EB 5.x where this will be the default in the PythonPackage EasyBlock